ert_shared/ensemble_evaluator/entity/snapshot.py

```diff
@@ -224,6 +257,23 @@ def get_job(self, real_id, stage_id, step_id, job_id):
             raise ValueError(f"No job with id {job_id} in {step_id}")
         return jobs[job_id]
 
+    def all_stages_finished(self, real_id):
+        real = self.get_real(real_id)
+        return all(stage["status"] == "Finished" for stage in real["stages"].values())
+
+    def all_steps_finished(self, real_id, stage_id):
+        stage = self.get_stage(real_id, stage_id)
+        return all(step["status"] == "Finished" for step in stage["steps"].values())
+
+    def get_successful_realizations(self):
+        return len(
+            [
+                real
+                for real in self.to_dict()["reals"].values()
+                if real["queue_state"] == "JOB_QUEUE_SUCCESS"
+            ]
+        )
```

Comment:
Shouldn't the number of successful realizations be based on the realization status instead?

Reply:
As far as I know, the realization, in this case, has no `"status" `.

Reply:
We can add status to the realization and check that one